crates/api/src/transport.rs

```diff
@@ -31,8 +31,11 @@ impl TxImpHnd {
 
     /// Call this when you receive or bind a new address at which
     /// this local node can be reached by peers
-    pub fn new_listening_address(&self, this_url: Url) {
-        self.handler.new_listening_address(this_url);
+    pub fn new_listening_address(
+        &self,
+        this_url: Url,
+    ) -> BoxFut<'static, K2Result<()>> {
+        self.handler.new_listening_address(this_url)
```

Comment:
The pattern for the handlers (for peer connect and recv data) is to return an error when the connection should be closed based on the data received (preflight checks in connect, and decoding or validation problems with recv). It doesn't make sense to me to have the implementer return an error into the library here (it's not going to shut down the whole endpoint because the client didn't like the address that was bound). What are your thoughts?